src/jvm/main/org/jetbrains/kotlinx/lincheck/trace/TracePoint.kt

```diff
@@ -370,22 +384,40 @@ internal sealed interface ReturnedValueResult {
     data class ValueResult(val valueRepresentation: String, val valueType: String): ReturnedValueResult
     
     // Holds any needed data to construct result lines
-    data class ActorResult(
+    interface ActorResult: ReturnedValueResult {
         // representation
-        val resultRepresentation: String,
+        val resultRepresentation: String
         
         // Needs to be shown next to the actor line as `actor(): ...`
-        val showAtBeginningOfActor: Boolean = true,
+        val showAtBeginningOfActor: Boolean
         
         // Needs to be shown after last event in actor as `result: ...`
-        val showAtEndOfActor: Boolean = true,
-        
-        // Is true if actor is hung, prevents empty actors from appearing
-        val isHung: Boolean = false,
-        
-        // For idea plugin
-        val exceptionNumber: Int = -1,
-    ): ReturnedValueResult
+        val showAtEndOfActor: Boolean
+    }
+    
+    data object ActorHungResult: ActorResult {
+        override val resultRepresentation = "<hung>"
+        override val showAtBeginningOfActor = true
+        override val showAtEndOfActor = false
+    }
+
+    data class ActorValueResult(override val resultRepresentation: String): ActorResult {
+        override val showAtBeginningOfActor = true
+        override val showAtEndOfActor = true
+    }
+    
+    data object ActorVoidResult: ActorResult {
```

Comment:
We have a duplication writ to `VoidResult`, `ValueResult`, etc. 
Maybe we don't need `ActorResult` as separate sub-class? 
Can we just re-use regular `ReturnedValueResult`, and add a boolean flag `isActorResult` ?

Reply:
I agree there is some redundancy. However, we would also have to add ways to distinguish exception result and hung result from value results. As this influences adding results nodes during trace flattening.

Additionally, when we add an `actorData` field to `VoidResult` (and others) would require us to create seperate result instances for each method (which are currently reusing the same objects in case of void or no result). To solve this we could have one `VoidResult` instance for all methods and one for all actors. But then we would be back at the the same duplication problem stated in your comment.